mitosheet/src/mito/components/footer/SheetTabContextMenu.tsx

```diff
@@ -1,27 +1,33 @@
 // Copyright (c) Saga Inc.
 
 import React, { useEffect } from 'react';
-import { MitoAPI,  getRandomId } from '../../api/api';
-import { DFSource, GraphDataDict, GraphID, SheetData, UIState } from '../../types';
+import { MitoAPI } from '../../api/api';
+import { DFSource, EditorState, GraphData, GraphDataArray, GraphID, GraphParamsBackend, SheetData, UIState } from '../../types';
 import Dropdown from '../elements/Dropdown';
 import DropdownItem from '../elements/DropdownItem';
 import DropdownSectionSeperator from '../elements/DropdownSectionSeperator';
 import { ModalEnum } from '../modals/modals';
-import { getDefaultGraphParams } from '../taskpanes/Graph/graphUtils';
 import { TaskpaneType } from '../taskpanes/taskpanes';
+import { openGraphSidebar } from '../taskpanes/Graph/graphUtils';
+import { GraphType } from '../taskpanes/Graph/GraphSetupTab';
 
 /*
     Helper function for finding all of the graph tab names
     that are created from a given sheet index
 */
-export const getGraphTabNamesAndIDsFromSheetIndex = (sheetIndex: number, graphDataDict: GraphDataDict): ({graphTabName: string, graphID: GraphID})[] => {
+export const getGraphTabNamesAndIDsFromSheetIndex = async (sheetIndex: number, graphDataArray: GraphDataArray, mitoAPI: MitoAPI): Promise<({graphTabName: string, graphID: GraphID})[]> => {
     // Filter to only grapsh with the sheetIndex, and then get a list of the graph tab names
-    const filteredGraphDataJSON: GraphDataDict = Object.fromEntries(Object.entries(graphDataDict || {}).filter(([, graphData]) => {
-        return graphData.graphParams.graphCreation.sheet_index === sheetIndex
-    }))
-
+    const response = await mitoAPI.getAllParamsForType<GraphParamsBackend>('graph')
+    const allGraphParams: GraphParamsBackend[] | undefined = 'error' in response ? undefined : response.result;
+    if (allGraphParams === undefined) {
+        return []
+    }
+    const filteredGraphDataJSON: GraphDataArray = graphDataArray.filter((graphData: GraphData, index: number) => {
+        return allGraphParams[index].graph_creation.sheet_index === sheetIndex
+    })
```

Comment:
I think we only want the _most recent_ params on this graph, no? Otherwise, if you:
1. Make a graph off df1
2. Change it to df2
3. Delete df1

It will incorrectly identify this graph as worthy of deleting.